enhancements/monitoring/monitoring-config.md

```diff
@@ -0,0 +1,307 @@
+        Add example CustomResource: config/examples/clustermonitoringoperator.yaml
+    - Client codegen 
+    - Reconcile logic: https://github.com/openshift/cluster-monitoring-operator/pull/2350
+    - Add decoupling ConfigMap / CustomResource:
+        Controller logic is strongly dependant of *manifests.Config struct.
+        
+
+### Example configuration
+
+
+#### current ConfigMap
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    telemeterClient:
+      enabled: false
+    prometheusK8s:
+      volumeClaimTemplate:
+        metadata:
+          name: prometheus-data
+          annotations:
+            openshift.io/cluster-monitoring-drop-pvc: "yes"
+        spec:
+          resources:
+            requests:
+              storage: 20Gi
+```
+
+### CRD
+
+```
+apiVersion: clustermonitoring.config.openshift.io
+kind: ClusterMonitoring
+metadata:
+  name: cluster
+  namespace: openshift-monitoring
+spec:
+  prometheusK8s:
+    volumeClaimTemplate:
+      metadata:
+        name: prometheus-data
+        annotations:
+          openshift.io/cluster-monitoring-drop-pvc: "yes"
+      spec:
+        resources:
+          requests:
+            storage: 20G
+```
+
+
+### Workflow Description
+
+See example in the [Proposal](#Proposal) for details.
+
+### API Extensions
+
+Add two new CRD's:
+- ClusterMonitoring
+- UserWorkloadClusterMonitoring
+
+### Topology Considerations
+
+Will function identically in all topologies.
+
+#### Hypershift / Hosted Control Planes
+No special considerations.
+
+#### Standalone Clusters
+No special considerations.
+
+#### Single-node Deployments or MicroShift
+No special considerations.
+
+### Implementation Details/Notes/Constraints
+
+
+### Migration path
+
+Migration path it will be split in three phases
+
+- Phase 1: Pre-GA (Gradual Migration for Developers)
+  - In this phase only developers are expected to use CRD.
+  - Both ConfigMap and CRD will coexist for a while until the transition is complete.
+  - Work with a Feature flag.
+  - Two options: ConfigMap or CRD.
+      Preference is to use ConfigMap over CRD during this phase.
+  - Behavior: If an entry exists in the ConfigMap, ignore the CRD.
+      If there’s no entry in the ConfigMap, use the CRD.
+  - Merge Process (only for phase 1)
+      Top-level fields (e.g., AlertmanagerMainConfig *AlertmanagerMainConfig || spec.AlertmanagerMainConfig):
+      If a field is nil in the ConfigMap, use the CRD; otherwise, stick with the ConfigMap.
+
+- Phase 2: Transition (User Control)
+  Discuss whether to keep dual support for a couple of releases.
+  - New Logic:
+    Allow users to choose the source of configuration (ConfigMap or CRD, warn about CRD preference).
+    If a CRD exists, prioritize it over the ConfigMap.
+    Add logic for users to manually switch to CRD if desired.
```

Comment:
What would the workflow look like for this? Will you build a tool that converts the configmap to CRD for them?

Reply:
at this point we dont have a merge process, we should choose between CR or configmap.

Reply:
We can add some tips to do the migration, but  user should read CR API and configure every type. We will work to improve fields as in

` This approach allows for a thorough review, improvement, and modification of each ConfigMap component to ensure it aligns with OpenShift API standards. As part of this process, types will be modified, outdated elements will be removed and names and configurations will be refined.`

Then users should read the release notes, the API, the tips or whatever and create a new CR.

Reply:
> We can add some tips to do the migration, but user should read CR API and configure every type.

You will have lazy users, who do not want to do this, and therefore will stay on the configmap forever. Is there anything we can do to provide a tool that would convert the configuration for them, as that possibly unblock you being able to remove configmap support

Reply:
I don't think is a good idea, CR is the standard way to defined resources and users are habituated to deal with it. To add an extra tool to will add extra work and could be error prone with different versions/releases.

Reply:
Maybe not an extra tool then, but you will need to provide a migration path before you can remove support for the configmap.

That may well just be documentation, but you'll likely need the end of that documentation to result in deleting the configmap, so that you can, in the future, block upgrades for anyone who has not yet migrated.

I would also consider folks like service delivery, do they use the configmap today? What would migration look like for them, can they easily script the migration?

Reply:
Add new info, please close discussion or add suggestion in order to clean the PR. Thanks!